src/models/v2/channel.ts

```diff
@@ -53,8 +54,9 @@ export class Channel extends BaseModel<v2.ChannelObject, { id: string, address:
   operations(): OperationsInterface {
     const operations: OperationInterface[] = [];
     ['publish', 'subscribe'].forEach(operationAction => {
+      let id =  this._json[operationAction as 'publish' | 'subscribe'] && (this._json[operationAction as 'publish' | 'subscribe'] as OperationObject).operationId || this.meta().id + "_" + operationAction;
       this._json[operationAction as 'publish' | 'subscribe'] && operations.push(
-        this.createModel(Operation, this._json[operationAction as 'publish' | 'subscribe'], { id: operationAction, action: operationAction, pointer: `${this._meta.pointer}/${operationAction}` }),
+        this.createModel(Operation, this._json[operationAction as 'publish' | 'subscribe'], { id: id, action: operationAction, pointer: `${this._meta.pointer}/${operationAction}` }),
```

Comment:
In JS (in Go as I recall you probably have the same feature) if the key you pass to an object, exists in the "shape" of that object then you can an avoid declaring key, so:

```suggestion
        this.createModel(Operation, this._json[operationAction as 'publish' | 'subscribe'], { id, action: operationAction, pointer: `${this._meta.pointer}/${operationAction}` }),
```

is enough :)